load_config.go

```diff
@@ -0,0 +1,50 @@
+// +build !darwin
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+func loadConfig(profile string) (*Config, error) {
+	dir, err := os.UserConfigDir()
```

Comment:
Hi there,

Allow me to jump in.
Does it also make sense to create a function to return the config directory?

```
func configDir() (string, error) (
    if os.GOOS == "darwin" {
        return os.UserHomeDir()
    }

    return os.UserConfigDir()
}
```

Then we don't need separate files per platform.

Reply:
Great call, updated.

Reply:
homedir is not required at least?

Reply:
`homedir` or similar is required for macOS - the example code above is copypasta and doesn't actually work as is.

Again, `os.UserHomeDir()` returns `$HOME/Library/Application Support` and not `~/` on macOS.

Reply:
Hmm, it's strange. On my machine, `os.UserHomeDir()` returns the same value as `$HOME`.
I am on go `1.20`.

Reply:
Sorry, I misread `os.UserHomeDir` for `os.UserConfigDir`. The latter is what algia uses today and causes the problem, but you're totally correct that `os.UserHomeDir` returns the homedir on macOS.

Updated the PR again to only us package `os`.